Libs/Core/ctkJobScheduler.cpp

```diff
@@ -312,6 +326,9 @@ void ctkJobScheduler::stopAllJobs(bool stopPersistentJobs)
     job->setStatus(ctkAbstractJob::JobStatus::Stopped);
     worker->cancel();
     }
+
+  this->waitForDone(2000);
```

Comment:
There are two very different use cases: shutdown and stopping all current jobs.

The current implementation looks more like a shutdown function (especially when stopPersistentJobs is true).

For stopping all current jobs, we could add a `stopAllJobs()` convenience function that would get the list of running jobs and call `stopJobsByJobUIDs()` for each. It would not block the execution (unless we add an optional parameter to block it, but it should be rarely used if ever).

Reply:
see last commit